lib/trackler/doc_file.rb

```diff
@@ -0,0 +1,92 @@
+class DocFile
+  DEFAULT_IMAGE_PATH = "/docs/img"
+
+  def self.find(basename:, track_dir:, image_path:)
+    dir = File.join(track_dir, "docs")
+
+    [
+      MarkdownFile.new(basename: basename, docs_dir: dir, image_path: image_path),
+      OrgmodeFile.new(basename: basename, docs_dir: dir, image_path: image_path),
+      NullFile.new(basename: basename, docs_dir: dir, image_path: image_path),
+    ].detect(&:acceptable?)
+  end
+
+  attr_reader :basename, :dir, :image_path
+  def initialize(basename:, docs_dir:, image_path:)
+    @basename = basename
+    @dir = docs_dir
+    @image_path = image_path
+  end
+
+  def to_md
+    transform(body).gsub(img_src, img_dst)
+  end
+
+  def name
+    "%s.%s" % [basename, extension]
+  end
+
+  def extension
+    "md"
+  end
+
+  def acceptable?
+    File.exist?(path)
+  end
+
+  private
+
+  def body
+    File.read(path)
+  end
+
+  def path
+    "%s/%s" % [dir, name]
+  end
+
+  def img_src
+    Regexp.new("]\\(%s" % DEFAULT_IMAGE_PATH)
+  end
+
+  def img_dst
+    "](%s" % image_path.gsub(Regexp.new("/$"), "")
+  end
+
+  def transform(s)
+    s
+  end
+end
+
+class OrgmodeFile < DocFile
+  def transform(s)
+    Orgmode::Parser.new(s).to_markdown
+  end
+
+  def extension
+    "org"
+  end
+end
+
+class MarkdownFile < DocFile
+end
+
+class NullFile
+  def initialize(basename:, docs_dir:, image_path:)
+  end
+
+  def to_md
+    ""
+  end
+
+  def name
+    ""
+  end
+
+  def extension
+    "md" # default for missing files
```

Comment:
Probably both.

It should be tested.
Having 100% test coverage is the easiest way to be sure that everything is tested and when adding new code easily checking that that code is tested, therefore even lines with "no risk" should be tested. 

It has risk.
Since the method is simple the test is easy to write.
If this line changes the class will behave differently.
I was doing some exploratory refactoring this code and ended up with a version that due to some changes in the inheritance structure did not return "md", this was not identified by the unit tests for this class.